### PR TITLE
Replace distutils with pkg_resources in __init__.py  due to deprecation

### PR DESCRIPTION
**Pull request by gareth-j**
This fixes the deprecation warning that was shown due to the deprecation of distutils. This replaces the use of `LooseVersion` with `pkg_resources.parse_version` in `__init__.py`.

### CODE REVIEW

**Review by davidhassell — COMMENTED** [review]
Hi @gareth-j,

Thanks for this! We have migrated away from LooseVersion in the other NCAS-CMS cf packages, but clearly not this one. I have one question, which perhaps you and/or @sadielbartholomew could answer - which is "better" (whatever that means): `pkg_resources.parse_version` as used here, or `packaging.version.Version` as used in `cf-python` and `cfdm`. I completely don't mind which we use, but it would be good to harmonise the approach across all three libraries, i.e. either using parse_version in the other two, or using Version here. We should go with what ever you and @sadielbartholomew agree - is that OK?

**Comment by gareth-j** [comment]
From what I understand `pkg_resources` is provided by `setuptools` that most users should have within their environment whereas `packaging` is a separate package. I did a quick test by creating a new virtual environment and `pkg_resources` was available after creation but `packaging` wasn't. I'm happy to modify this PR to use `packaging` though if that's chosen.

**Comment by davidhassell** [comment]
I'm always keen to reduce dependencies, so would be happy to use `pkg_resources` across the board. @sadielbartholomew, is that OK with you?

**Comment by davidhassell** [comment]
Hi - I'm keen to create 3.3.5 later this week, and in the absence of further thoughts, I shall merge this PR very soon ...

**Comment by sadielbartholomew** [comment]
Sorry all, I missed this due to distractions from a few conferences, I will take a look in a moment...

**Comment by sadielbartholomew** [comment]
Hi both, sorry again for the late response.

> We have migrated away from LooseVersion in the other NCAS-CMS cf packages, but clearly not this one.

References are PRs NCAS-CMS/cf-python#324 and NCAS-CMS/cfdm#173, just to include quick links in this thread.

> I have one question, which perhaps you and/or @sadielbartholomew could answer - which is "better" (whatever that means): pkg_resources.parse_version as used here, or packaging.version.Version as used in cf-python and cfdm.

When I opened NCAS-CMS/cf-python#324 I made some notes, anticipating I might forget the details on this. We are advised by the deprecation warning to use the latter:

> `DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.`

so, both given that direct pointer and to be consistent with cf-python and cfdm which have already been fixed with regards to the deprecation, as David states he would also like, I think we should use that instead.

However, note a side-effect to this choice is that we need to add `packaging>=20.0` as a new dependency with this choice, as done and noted on the PRs for `cf` and `cfdm`.

So overall: @gareth-j, thanks for raising the underlying issue here and your work so far on it. Would you mind converting this PR to use `packaging.version.Version` instead of `pkg_resources.parse_version`, for the reasons we've covered in our comments? You can use #324 as a template. You will also need to add `packaging` to `requirements.txt`. Sorry that we want an alternative approach, I guess if you'd opened an Issue to cover this first we could have worked this out before you put up the PR. It shouldn't be much work or take much time at all to convert, but if you'd rather not or don't have the time, either David or myself can step in to do this. Thanks.

**Comment by sadielbartholomew** [comment]
The linting check has failed but based on my comment above we can ignore that for now...

**Comment by davidhassell** [comment]
Hi Sadie - thanks you for your careful review, and reminding me of the history - I support your suggestions.

File: Changelog.rst [hunk 1]
```diff
@@ -1,3 +1,12 @@
+version 3.3.5
+-------------
+----
+
+**xxxx-xx-xx**
+
+* Remove calls to ``LooseVersion`` from ``distutils`` due to deprecation of that package,
```
Comment (sadielbartholomew) [inline Changelog.rst:7]: As a general comment, realising that this message will likely change as we convert the approach used, ideally keep the line length shorter for the message in line with the approximate maximum for the rest of this file. Thanks.

**Comment by sadielbartholomew** [comment]
@gareth-j: in the interests of time, because we want to make a new release of this library ASAP, I have updated the PR in line with my previous comments. We wanted to include your PR and commit as a thanks for your work on this, hence my pushing new commits rather than closing this and doing a separate PR.

@davidhassell please can you give this a quick review, then we can merge and prepare the new release.

Thanks both!

**Review by davidhassell — APPROVED** [review]
Great, thanks @sadielbartholomew and @gareth-j

**Comment by gareth-j** [comment]
Sorry I missed your comments @sadielbartholomew! And thanks for merging my PR!

**Pull request merged**